[rollout-operator] Enable setting containers.extraArgs

meinenec opened the pull request:
This PR is an update of https://github.com/grafana/helm-charts/pull/3473

I came across this while attempting to deploy the [Loki Helm chart](https://github.com/grafana/loki/tree/main/production/helm/loki) and found that I needed to set `args` for deploying with Mimir.

This will enable users to specify additional arguments while preserving the existing default.

56quarters commented on charts/rollout-operator/values.yaml:
```diff
@@ -51,6 +51,9 @@ securityContext: {}
   # runAsNonRoot: true
   # runAsUser: 1000
 
+# -- List of additional cli arguments to configure agent-operator (example: `--log.level`)
```
```suggestion
# -- List of additional cli arguments to configure rollout-operator (example: `--log.level=info`)
```

56quarters commented on charts/rollout-operator/README.md:
```diff
@@ -40,6 +40,7 @@ It is not a highly available application and runs as a single pod.
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
 | affinity | object | `{}` |  |
+| extraArgs | list | `[]` | List of additional cli arguments to configure agent-operator (example: `--log.level`) |
```
```suggestion
| extraArgs | list | `[]` | List of additional cli arguments to configure rollout-operator (example: `--log.level=info`) |
```

56quarters reviewed (approved):
LGTM except for some copy/paste issues.

56quarters commented:
Can you add a note in the `CHANGELOG` about this as well? Thanks!

meinenec commented:
I've addressed the comments. Thank you for catching them!

> Can you add a note in the `CHANGELOG` about this as well? Thanks!

Apologies, I don't see a CHANGELOG for the rollout-operator, where would you like the note to be added?

56quarters reviewed (approved):
> I've addressed the comments. Thank you for catching them!
> 
> > Can you add a note in the `CHANGELOG` about this as well? Thanks!
> 
> Apologies, I don't see a CHANGELOG for the rollout-operator, where would you like the note to be added?

:facepalm: My bad, I was mistaken. LGTM, thanks again!

andyasp commented on charts/rollout-operator/README.md:
```diff
@@ -40,6 +40,7 @@ It is not a highly available application and runs as a single pod.
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
 | affinity | object | `{}` |  |
+| extraArgs | list | `[]` | List of additional cli arguments to configure rollout-operator (example: `--log.level=info`) |
```
```suggestion
| extraArgs | list | `[]` | List of additional CLI arguments to configure rollout-operator (example: `--log.level=info`) |
```
DCO check looks stuck, adding a suggestion to see if it helps.

andyasp commented on charts/rollout-operator/values.yaml:
```diff
@@ -51,6 +51,9 @@ securityContext: {}
   # runAsNonRoot: true
   # runAsUser: 1000
 
+# -- List of additional cli arguments to configure rollout-operator (example: `--log.level=info`)
```
```suggestion
# -- List of additional CLI arguments to configure rollout-operator (example: `--log.level=info`)
```
